scaleway/resource_compute_instance_server.go

```diff
@@ -363,7 +380,44 @@ func resourceScalewayComputeInstanceServerUpdate(d *schema.ResourceData, m inter
 		updateRequest.EnableIPv6 = utils.Bool(d.Get("enable_ipv6").(bool))
 	}
 
-	_, err = instanceApi.UpdateServer(updateRequest)
+	volumes := map[string]*instance.VolumeTemplate{}
+	volumes["0"] = &instance.VolumeTemplate{ID: d.Get("root_volume.0.volume_id").(string), Name: getRandomName("vol")} // name is ignored by the API, any name will work here
+
+	if raw, ok := d.GetOk("additional_volume_ids"); d.HasChange("additional_volume_ids") && ok {
```

Comment:
What if you have additional_volume_ids but they did not change ....
With this condition you don't include them and they will be detached